File: src/main/java/com/bdev/hogwarts_api/service/events/EventService.java
package com.bdev.hogwarts_api.service.events;

import com.bdev.hogwarts_api.data.dto.events.Event;
import com.bdev.hogwarts_api.data.dto.events.EventType;

import java.util.List;
import java.util.Optional;

public interface EventService {
    List<Event> getAll();
    Optional<Event> getLatestEvent(EventType eventType);
    Optional<Event> getById(long id);
    boolean exists(long id);
    long create(Event speakingClub);
    void update(Event speakingClub);
    void delete(long id);
}

File: src/main/java/com/bdev/hogwarts_api/service/events/EventService.kt
package com.bdev.hogwarts_api.service.events

import com.bdev.hogwarts_api.data.dto.events.Event
import com.bdev.hogwarts_api.data.dto.events.EventType

interface EventService {
    fun getAll(): List<Event>
    fun getLatestEvent(eventType: EventType): Event?
    fun getAllFutureEvents(eventType: EventType): List<Event>
    fun getById(id: Long): Event?
    fun exists(id: Long): Boolean
    fun create(speakingClub: Event): Long
    fun update(speakingClub: Event)
    fun delete(id: Long)
}
